app/models/order.rb

```diff
@@ -0,0 +1,48 @@
+class Order < ApplicationRecord
+  belongs_to :cart
+  enum status: [:ordered,:completed,:cancelled,:paid]
+  TYPE={ordered:"ordered",completed:"completed",cancelled:"cancelled",paid:"paid",all:"all"}
+
+  after_initialize do
+    if self.new_record?
+      self.status ||= :ordered
+    end
+  end
+
+  def self.place_order(cart)
+    total=Order.total_order_price(cart)
+    order=Order.new(cart_id:cart.id,user_id:cart.user_id,total_amount:total)
+    if order.save!
```

Comment:
why use two statements.

Reply:
because If I use create method in place of new and save then how will I know that order is saved or not?